website/docs/pubsub.md

```diff
@@ -0,0 +1,188 @@
+---
+id: pubsub
+title: Pubsub
+---
+
+`Feature State`: Gatekeeper version v3.13+ (alpha)
+
+> ❗ This feature is still in alpha stage, so the final form can still change (feedback is welcome!).
+
+## Motivation
+
+Prior to this feature, there were two ways to get audit violations. One is to look at constraints status and the other is to look at audit pod logs and get the logged audit violations. Both of these approach have limitations as described below.
+
+Limitations of getting audit violations from constraint status:
+
+- To reduce in memory consumption of audit and avoid violating etcd limit of 1.5MB per resource, gatekeeper only reports up-to 500 violations on constraint template. In other words, one might not get all the violations from looking at constraint.
+
+Limitations of getting audit violations from audit logs:
+
+- It could be difficult to parse audit logs and look for violation messages as violation logs would be scrambled together with other log statements. Additionally, when there are huge number of violations, the pod logs might get pruned and it might not be possible to get the logs for whole audit. In other words, it is possible to look at logs to fetch audit violation and still not get all the violations that were caught during audit.
```

Comment:
```suggestion
- It could be difficult to parse audit pod logs to look for violation messages, as violation logs would be scrambled together with other log statements. Additionally, when there are huge number of violations, it’s possible to miss part of the log as the pod logs get rotated. As a result, it is possible to look at logs to fetch audit violation and still not get all the violations that were caught during audit.
```